### ISSUE TITLE
User terminated GeoDa: LISA using time grouped variable, which takes about 10 minutes to run

### ISSUE THREAD

**Issue by GeoDaTester** [GeoDaCenter/geoda#1369]
From: 
 Details:
os: 1-10-13
vs: 1-12-1-47
/Applications/GeoDa.app/Contents/MacOS/../Resources/logger.txt
 GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Check auto update:
AutoUpdate::CheckUpdate()
AutoUpdate::GetCheckList()
AutoUpdate::ReadUrlContent()
Entering ConnectDatasourceDlg::OnOkClick
GdaFrame::OpenProject()
/Users/noahschouela/Desktop/Final_ISDS/Mexic_Mun_2456_Weights.gda
Entering Project::Project (existing project)
/Users/noahschouela/Desktop/Final_ISDS/Mexic_Mun_2456_Weights.gda
Project::SetProjectFullPath()
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/Users/noahschouela/Desktop/Final_ISDS/Mexic_Mun_2456.shp
Project::GetSpatialReference()
Exiting Project::Project
Click GdaFrame::InitWithProject()
Open TableFrame.
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
Open project file:/Users/noahschouela/Desktop/Final_ISDS/Mexic_Mun_2456_Weights.gda
MapCanvas::~MapCanvas()
Open LisaMapFrame (OnOpenUniLisa).
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
HomEbs
VariableSettingsDlg::GetWeightsId() weight: Mexic_Mun_2456_q2inclusive
VariableSettingsDlg::GetWeightsId() weight: Mexic_Mun_2456_q2inclusive
Entering LisaCoordinator::LisaCoordinator().
Entering LisaCoordinator::SetSignificanceFilter()
Exiting LisaCoordinator::SetSignificanceFilter()
Entering LisaCoordinator::InitFromVarInfo()
Entering LisaCoordinator::DeallocateVectors()
Exiting LisaCoordinator::DeallocateVectors()
Entering LisaCoordinator::AllocateVectors()
Exiting LisaCoordinator::AllocateVectors()
Entering LisaCoordinator::StandardizeData()
Exiting LisaCoordinator::StandardizeData()
Entering LisaCoordinator::CalcLisa()
Exiting LisaCoordinator::CalcLisa()
Entering LisaCoordinator::CalcPseudoP()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP()
Exiting LisaCoordinator::InitFromVarInfo()
Exiting LisaCoordinator::LisaCoordinator().
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
In LisaCoordinator::registerObserver()
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
In LisaCoordinator::registerObserver()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
In OnRanOtherPer()
99999
Entering LisaCoordinator::CalcPseudoP()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()

**Comment by lixun910** [comment]
Hi Noah,

Thanks for reporting this issue! Is this a crash when you try to do permutation (99999) in a univariate LISA map? Can you share your date (Mexic_Mun_2456.shp/shx/dbf) with us, so we can replicate and fix this issue? We will keep it confidential.

Best,

Xun Li,  Ph.D.

Assistant Director for Data Science
Center of Spatial Data Science
University of Chicago
http://spatial.uchicago.edu



On Nov 29, 2017, at 2:13 PM, GeoDaTester <notifications@github.com<mailto:notifications@github.com>> wrote:


From:
Details:
os: 1-10-13
vs: 1-12-1-47
/Applications/GeoDa.app/Contents/MacOS/../Resources/logger.txt
GdaFrame::ShowOpenDatasourceDlg()
ConnectDatasourceDlg::InitSamplePanel()
Check auto update:
AutoUpdate::CheckUpdate()
AutoUpdate::GetCheckList()
AutoUpdate::ReadUrlContent()
Entering ConnectDatasourceDlg::OnOkClick
GdaFrame::OpenProject()
/Users/noahschouela/Desktop/Final_ISDS/Mexic_Mun_2456_Weights.gda
Entering Project::Project (existing project)
/Users/noahschouela/Desktop/Final_ISDS/Mexic_Mun_2456_Weights.gda
Project::SetProjectFullPath()
Project::CommonProjectInit()
Entering Project::InitFromOgrLayer
Datasource name:
/Users/noahschouela/Desktop/Final_ISDS/Mexic_Mun_2456.shp
Project::GetSpatialReference()
Exiting Project::Project
Click GdaFrame::InitWithProject()
Open TableFrame.
Open MapFrame.
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
Open project file:/Users/noahschouela/Desktop/Final_ISDS/Mexic_Mun_2456_Weights.gda
MapCanvas::~MapCanvas()
Open LisaMapFrame (OnOpenUniLisa).
Open VariableSettingsDlg
Click VariableSettingsDlg::OnOkClick:
HomEbs
VariableSettingsDlg::GetWeightsId() weight: Mexic_Mun_2456_q2inclusive
VariableSettingsDlg::GetWeightsId() weight: Mexic_Mun_2456_q2inclusive
Entering LisaCoordinator::LisaCoordinator().
Entering LisaCoordinator::SetSignificanceFilter()
Exiting LisaCoordinator::SetSignificanceFilter()
Entering LisaCoordinator::InitFromVarInfo()
Entering LisaCoordinator::DeallocateVectors()
Exiting LisaCoordinator::DeallocateVectors()
Entering LisaCoordinator::AllocateVectors()
Exiting LisaCoordinator::AllocateVectors()
Entering LisaCoordinator::StandardizeData()
Exiting LisaCoordinator::StandardizeData()
Entering LisaCoordinator::CalcLisa()
Exiting LisaCoordinator::CalcLisa()
Entering LisaCoordinator::CalcPseudoP()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP()
Exiting LisaCoordinator::InitFromVarInfo()
Exiting LisaCoordinator::LisaCoordinator().
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
In LisaCoordinator::registerObserver()
MapCanvas::MapCanvas()
MapCanvas::ChangeMapType()
MapCanvas::VarInfoAttributeChange()
MapCanvas::CreateAndUpdateCategories()
In LisaCoordinator::registerObserver()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Entering LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
Exiting LisaCoordinator::CalcPseudoP_range()
In OnRanOtherPer()
99999
Entering LisaCoordinator::CalcPseudoP()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()
Exiting LisaCoordinator::CalcPseudoP_threaded()
Entering LisaCoordinator::CalcPseudoP_threaded()

**Comment by lixun910** [comment]
That explains it! 
I kept quitting the application after about 5 mins or so because Geoda was “not responding.”
Thanks!

--
Hi Noah,

Thank you very much! 

I was able to use your data and run 99,999 permutations on the LISA of HomEbs for 2010. But, it took a long time: about 10 minutes to run. 
The reason it is very slow is because GeoDa is trying to run 99,999 permutations on all 6 time steps (2006-2011), and user can navigate the LISA maps in different time step using “Time->Time Player” (see the screenshot):

<img width="604" alt="screen shot 2017-11-29 at 4 24 48 pm" src="https://user-images.githubusercontent.com/2423887/33415407-9104fbd4-d553-11e7-87da-7cde7a45e9e2.png">
